srcCpp/perftest_publisher.cxx

```diff
@@ -123,6 +128,51 @@ int perftest_cpp::Run(int argc, char *argv[])
     }
 }
 
+bool perftest_cpp::set_main_thread_priority()
+{
+    int priority = perftest_cpp::threadPriorities.main;
+
+#ifdef RTI_WIN32
+    unsigned long error;
+
+    if (!SetPriorityClass(GetCurrentProcess(), REALTIME_PRIORITY_CLASS)){
+        error = GetLastError();
+            printf("Fail to set main thread Class to real time, ERROR: %d\n", error);
+        return false;
+    }
+
+    if (!SetThreadPriority(GetCurrentThread(), priority)) {
+        error = GetLastError();
+        if (priority == error)
+            printf("The thread is already running with priority ERROR: %d\n", error);
+        else {
+            printf("Fail to set main thread priority, ERROR: %d\n", error);
+        }
+        return false;
+    }
+#else
```

Comment:
This might lead to issues. I would do this instead:

```
#ifdef RTI_WIN32
  . . .
#elsif defined(RTI_LINUX) || defined(RTI_DARWIN) <- Maybe you can even use RTI_UNIX, but you will need
                                                            to check if this is standard for all UNIX.
 . . . 
#else
  <Add code saying that priorities are not supported for this platform>
#endif
```

Reply:
I'm gonna use RTI_UNIX

*"Implementations of the API are available on many Unix-like POSIX-conformant operating systems such as FreeBSD, NetBSD, OpenBSD, Linux, Mac OS X, Android and Solaris"*